transport/http/client.go

```diff
@@ -49,10 +50,37 @@ func (c ClientHandler) Handle(ctx context.Context, input interface{}) (
 
 	resp, err := c.client.Do(req.Build(ctx))
 	if err != nil {
-		return nil, metadata, err
+		err = &RequestSendError{Err: err}
+
+		// Override the error with a context canceled error, if that was canceled.
+		select {
+		case <-ctx.Done():
+			err = &smithy.CanceledError{Err: ctx.Err()}
+		default:
+		}
 	}
 
-	return &Response{Response: resp}, metadata, nil
+	return &Response{Response: resp}, metadata, err
+}
+
+// RequestSendError provides a generic request transport error.
+type RequestSendError struct {
+	Err error
+}
+
+// ConnectionError return that the error is related to not being able to send
+// the request.
+func (e *RequestSendError) ConnectionError() bool {
+	return true
+}
```

Comment:
Hmmm does this need to be smarter? This would cause all errors to be retried. As the retryer will retry all errors that implement ConnectionError and return true.

Reply:
I think that was the original intention. But will double check to make sure the intention here.